### ISSUE TITLE
Certain Forge mod mixins crashes when ran on GDLauncher due to its Java version

### ISSUE THREAD

**Issue by TelepathicGrunt** [gorilla-devs/GDLauncher#674]
Hello! I am a mod developer of the 1.16.3 mod Blame and I was trying to add extra info to the logs when mods try to load missing structure files. However, I have found a few mixins will crash GDLauncher and only GDLauncher. Crashlog looks this this: https://hatebin.com/qyncobosbk

The first mixin I tried that crashed GDLauncher.
https://github.com/TelepathicGrunt/Blame/blob/9b356c6cdf5b81a6c68d9077c31878f03dcecb70/src/main/java/com/telepathicgrunt/blame/mixin/JigsawPatternMixin.java#L35-L46

The second mixin to replace the first (the third is the same but but method is `Lnet/minecraft/world/gen/feature/jigsaw/SingleJigsawPiece;getBoundingBox(Lnet/minecraft/world/gen/feature/template/TemplateManager;Lnet/minecraft/util/math/BlockPos;Lnet/minecraft/util/Rotation;)Lnet/minecraft/util/math/MutableBoundingBox;` instead):
https://github.com/TelepathicGrunt/Blame/blob/be48a0699e508a44bdce660f03727f753cac27a7/src/main/java/com/telepathicgrunt/blame/mixin/SingleJigsawPieceMixin.java#L28-L38

All my other mixins works perfectly fine on GDLauncher. And the problematic mixins works fine with all other launchers. Someone said to try GDLauncher but with my own Java version so I changed GDLauncher's settings to use the jre1.8.0_261 java I had installed. And well, my mod worked with no startup crash at all.

So the issue seems to be due to the java version currently shipped with GDLauncher which is: java\8u242-b08\bin\java.exe
It may be a good idea to update the java or switch to 8u51 java to match vanilla's java version. I suspect as more and more mods use mixins on Forge, this issue is going to become more and more common with GDLauncher if it doesn't change the java that it uses.

I hope this helps!

The original bug report I gotten about this issue: https://github.com/TelepathicGrunt/Blame/issues/4

**Comment by TelepathicGrunt** [comment]
My friends just pointed out that GDLauncher should be shipping 8u51 java to match vanilla as one of the reasons Mojang stuck with it was because 8u60+ java may break Intel chips. So sticking with 8u51 should help close any issues reported to GDLauncher that stems from the java version itself. Though the mixin issue is quite odd and I'll try and ask around for how to handle that as mixin shouldn't be dependent on java versions I thought.
https://github.com/LWJGL/lwjgl/issues/119
https://github.com/MultiMC/MultiMC5/issues/1276

**Comment by CrankySupertoon** [comment]
Not to mention GDLauncher is using some super ancient version priopetorely hosted on Davide's Website. My Upstream fork Koala Launcher uses the latest Java 8 instead of davide's one.

This commit explains alot

https://github.com/KoalaDevs/KoalaLauncher/commit/699dc86a1aa1984892754b2e1aa84fc26855d772

Davide's link was used in the rewrite using it rehosted since Davide is extremly picky about that stuff.

GDLauncher's Java manifest is here 

https://cdn.assets-gdevs.com/openjdk8.json

the offical openjdk endpoints that are indentical and he copied from are here

https://api.adoptopenjdk.net/v2/latestAssets/releases/openjdk8?arch=x64&heap_size=normal&openjdk_impl=hotspot&type=jre

**Comment by jamierocks** [comment]
> My Upstream fork Koala Launcher uses the latest Java 8 instead of davide's one.

I believe you mean either your fork, or your downstream project - in any case, this issue isn't about GDLauncher not using the latest version - its about using too new of a version.

**Comment by furtabs** [comment]
I misinterpreted this post. I thought op was complaining about some old java

**Comment by TelepathicGrunt** [comment]
As I was attaching the jdk-8.0.252.09-hotspot to my dev environment, I noticed intellij marked import javafx.util.Pair; as non-existent. When I removed that code with the proper import com.mojang.datafixers.util.Pair;, the mod started working with openjdk and the others.

Turns out, the mixin I was using used javafx.util.Pair; and crashed. But I still don't know why using javafx.util.Pair; outside the mixin in my own code (with the problematic mixin removed) stopped the crash.

Nevertheless, this is 100% my fault and sorry about that! I'll close this issue. This is definitely a mistake I'll never make twice lol
